pkg/cli/admin/nodeimage/create.go

```diff
@@ -459,13 +489,124 @@ func (o *CreateOptions) waitForCompletion(ctx context.Context) error {
 				return false, nil
 			}
 
-			// Extract node-joiner tool exit code on completion
-			o.nodeJoinerExitCode, err = strconv.Atoi(w.String())
+			return true, nil
+		})
+
+	if err != nil {
+		return nil, fmt.Errorf("error caught while executing remote command: %w. Error output: %s", err, wErr.String())
```

Comment:
I suggest only add "Error output: %s" if wErr.String() is not empty. For example, I received this output when testing the command and terminating the node-joiner pod while it was in the middle of processing:
```
2024-11-19T20:00:17Z [node-image create] unexpected error caught while running the command, storing pod logs in report
2024-11-19T20:00:17Z [node-image create] command execution failed. Reason: error caught while executing remote command: Internal error occurred: error executing command in container: container is not created or running. Error output: 
2024-11-19T20:00:17Z [node-image create] Saving report file
```

Reply:
Given the exceptional condition I though it was useful to report both in any case (even if empty, just to be sure to not miss anything else). But if it's misleading we could remove (it in a subsequent pr)